### ISSUE TITLE
Installation issue - Unable to Identify CMake Compilers

### ISSUE THREAD

**Issue by asimon3210** [swarm-lab/ROpenCVLite#21]
Hello there! I'm trying to download ROpenCVLite (R version 3.6.1, CMake version 3.17, Rtools version 3.5), but I'm encountering the following error: 

`-- The CXX compiler identification is unknown
-- The C compiler identification is unknown
CMake Error at CMakeLists.txt:99 (enable_language):
The CMAKE_CXX_COMPILER:

c:/Rtools/usr/mingw_64/bin/g++.exe
is not a full path to an existing compiler tool.

Tell CMake where to find the compiler by setting either the environment
variable "CXX" or the CMake cache entry CMAKE_CXX_COMPILER to the full path
to the compiler, or to the compiler name if it is in the PATH.`

I checked to make sure R is able to find Rtools (find_rtools() returns TRUE), but when I run pkgbuild::rtools_path, it shows C:/Rtools/usr/bin, but the necessary Rtools compilers are in C:/Rtools/mingw_64/bin ( C:/Rtools/bin and  C:/Rtools/mingw_64/bin are also in my system path) 

Does anyone have any idea on how to fix this? Thanks in advance!

**Comment by sjmgarnier** [comment]
@asimon3210 Are you certain that the version of Rtools that you have is 3.5?

**Comment by asimon3210** [comment]
Thank you for your response! Yes, I'm sure - I installed it today after thinking that uninstalling CMake and Rtools and reinstalling would fix the issue

**Comment by sjmgarnier** [comment]
@sjmgarnier I'm a bit confused. You are saying that C:/Rtools/mingw_64/bin exists on your drive and your path, but that CMake doesn't seem to find it, right? Just to make sure, did you log in/out of your session after adding Rtools to the path? I don't think that Windows automatically reload the path after it's been modified.

**Comment by asimon3210** [comment]
> You are saying that C:/Rtools/mingw_64/bin exists on your drive and your path, but that CMake doesn't seem to find it, right?

Yes. And I made sure to close RStudio and reopen it before trying to install again. I'm sorry for not being clear. I'm pretty new to CMake and OpenCV.

**Comment by sjmgarnier** [comment]
@asimon3210 No, I meant log in and out of your Windows session, or even rebooting the system completely.

**Comment by asimon3210** [comment]
No, I actually haven't tried that. I'm currently using R on a virtual machine and only have a limited amount of time to be logged in each day, so I unfortunately can't try that now. Do you mind if I try this tomorrow once my session starts and update you on the results?

**Comment by sjmgarnier** [comment]
@asimon3210 Sure, no problem.

**Comment by asimon3210** [comment]
Hello @sjmgarnier! I logged in and out of my window session, and I'm still seeing the same error when trying to install ROpenCVLite. `pkg_build::rtools_path` is still reading as `"c:/Rtools/usr/bin"` even though the CMake compilers are in `C:/Rtools/mingw_64/bin`

**Comment by sjmgarnier** [comment]
@asimon3210 The only thing I can think of is that the path is not correctly defined. What's the result of ```Sys.getenv("PATH")```?

**Comment by asimon3210** [comment]
I get ` "c:\\Rtools\\bin;c:\\Rtools\\mingw_64\\bin;c:\\fftw;C:\\Windows\\system32;C:\\Windows;C:\\Windows\\System32\\Wbem;C:\\Windows\\System32\\WindowsPowerShell\\v1.0\\;C:\\Windows\\System32\\OpenSSH\\;C:\\Program Files\\Amazon\\cfn-bootstrap\\;C:\\Program Files\\R\\R-3.6.1\\bin\\x64;C:\\Program Files\\CMake\\bin;C:\\Program Files\\MATLAB\\R2020a\\bin;C:\\Users\\calderonlab\\AppData\\Local\\Microsoft\\WindowsApps;"`

**Comment by sjmgarnier** [comment]
@asimon3210 This is really strange. I can't reproduce the issue here, so I'm not sure what to do to help. I'm guessing there is no way I can access this machine remotely, right?

**Comment by asimon3210** [comment]
I don't think so, as this is a virtual machine that I'm using through my university. I could share my screen with you via Zoom if you think that would help

**Comment by sjmgarnier** [comment]
@asimon3210 We can possibly do that but I won't have time to meet before this weekend.

**Comment by asimon3210** [comment]
Hi @sjmgarnier , no need to worry - I managed to get around the issue by just moving the contents of my C:\Rtools folder into a new folder called 'usr'. I checked if the folder 'pkgbuild::rtools_path' was finding existed via Command Prompt and saw that it didn't exist. I don't know why R was finding Rtools there in the first place, but creating the folder seems to work. 

Thanks anyway for all your help! Sorry this issue was so confusing
